### PR TITLE
ShareDrawer: Share embed panel

### LINKED ISSUE #90494: Embed feature

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

- Adds "Share embed" feature for panels`

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/90494

**Special notes for your reviewer:**
Please, review https://github.com/grafana/grafana/pull/90549 first

| Share panel button  | Drawer content |
| ------------- | ------------- |
| image | image |

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx [hunk 1]
```diff
@@ -96,6 +97,19 @@ export function panelMenuBehavior(menu: VizPanelMenu, isRepeat = false) {
           dashboard.showModal(drawer);
         },
       });
+      subMenu.push({
+        text: t('share-panel.menu.share-embed-title', 'Share embed'),
```
Comment (evictorero) [inline public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx:101]: Is `Share embed` clear enough? In the other cases (Share Link, Share Snapshot) the Sharing action or resource is clear.
Maybe we could use "Embed Panel" in this case or something like that.

File: public/app/features/dashboard-scene/scene/keyboardShortcuts.ts [hunk 2]
```diff
@@ -57,6 +58,17 @@ export function setupKeyboardShortcuts(scene: DashboardScene) {
...
 
+        scene.showModal(drawer);
+      }),
+    });
+    keybindings.addBinding({
+      key: 'p e',
+      onTrigger: withFocusedPanel(scene, async (vizPanel: VizPanel) => {
+        const drawer = new ShareDrawer({
+          title: t('share-panel.drawer.share-embed-title', 'Embed'),
+          body: new SharePanelEmbedTab({ panelRef: vizPanel.getRef() }),
```
Comment (evictorero) [inline public/app/features/dashboard-scene/scene/keyboardShortcuts.ts:69]: This won't only represent a tab anymore. Does it make sense to rename it?

**Review by evictorero — APPROVED** [review]
Left some comments to think about but LGTM!

File: public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx (hunk 1 above)
Comment (juanicabanas) [reply public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx:101]: I think it makes sense when talking about the menu title, *but* I was wondering the same thing when it refers to the Drawer title, taking into account the Drawer is open with full screen height and maybe the user losses context and doesn't know what's been shared. Maybe "Share Embed Panel" would make sense. @natydej ?

File: public/app/features/dashboard-scene/scene/keyboardShortcuts.ts (hunk 2 above)
Comment (juanicabanas) [reply public/app/features/dashboard-scene/scene/keyboardShortcuts.ts:69]: I would leave the component as it is, as we're reusing the old component. Once we enabled this new redesign 100%, we need to make these kind of changes

File: public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx (hunk 1 above)
Comment (natydej) [reply public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx:101]: I think we should be consistent with the title option that we use on the dropdown, in this case "Share embed" should enough, people already know that they are coming from a panel flow.
Here are the [figma link](https://www.figma.com/design/AjvrAorYpnfSoMpU5kwf39/Adoption-of-Sharing?node-id=4560-57369&t=fl7JaNQzod3sIVbL-1) for designs
Screenshot 2024-08-02 at 10 58 11
Screenshot 2024-08-02 at 10 58 00

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. This is a friendly reminder that the Steady [11.2.x Release Branch](https://github.com/grafana/grafana/tree/v11.2.x) was cut this week, and that it includes the changes committed up to https://github.com/grafana/grafana/commit/b4126d3bce3627e9460e2db0e870c8d984be8e84. This PR was merged after it, so we wanted to let you know that you can add the `backport v11.2.x` label to to backport it and make it part of the next 11.2.x Patch Release, just in case you may have missed the news.

Thank you!

**Pull request merged**